### PR TITLE
Added stream creation to use Promises.

### PR DESCRIPTION
**Pull request by shampson**
Fix for [issue 50](https://github.com/w3c/webrtc-quic/issues/50)

### CODE REVIEW

**Comment by shampson** [comment]
https://rawgit.com/w3c/webrtc-quic/create-stream-promise/index.html

**Comment by shampson** [comment]
@aboba I merged with the cs.html and made a couple changes. To answer your notes:
1. Fixed
2. Looks like you fixed this.
3. The promise is resolved when the transport state transitions to "connected". Is there a case where the STREAM frame can't be sent when the transport is connected? This was already a problem previously, regardless of the streams being returned in a promise. I suppose in this case your readable/writable states would never transition to true.

**Comment by shampson** [comment]
Easy viewing:
https://rawgit.com/w3c/webrtc-quic/create-stream-promise/cs.html

File: cs.html [hunk 1]
```diff
@@ -210,39 +211,49 @@ <h2>Methods</h2>
               steps:</p>
               <ol>
                 <li>
-                  <p>Let <var>transport</var> be the <code><a>QuicTransportBase</a></code>
+                  <p>Let <var>transport</var> be the <code><a>QuicTransport</a></code>
```
Comment (pthatcherg) [inline cs.html]: Shouldn't this be QuicTransportBase?

File: cs.html [hunk 2]
```diff
@@ -253,38 +264,51 @@ <h2>Methods</h2>
...
                   on which <code>createSendStream</code> is invoked.</p>
```
Comment (pthatcherg) [inline cs.html]: Shouldn't this be QuicTransportBase?

**Comment by shampson** [comment]
Note - I need to add the streams to the internal slots properly, and properly replace QuicTransport-->QuicTransportBase

File: cs.html (hunk 1 above)
Comment (shampson) [reply cs.html]: Done.

File: cs.html (hunk 2 above)
Comment (shampson) [reply cs.html]: Done.

**Comment by shampson** [comment]
I made some fixes:
- Added back procedure for adding a stream to the transport slots. This was lost in the merge.
- Changed "fulfill" to "resolve"
- Update QuicTransport to QuicTransportBase

@aboba @pthatcherg You might want to give this a final look, but it should be ready to merge.

**Pull request merged**